### ISSUE TITLE
Server not putting in cache conversations messages 

### ISSUE THREAD

**Issue by Rob174** [nsmaniotto/Distributed-Chat-System#14]
Scenario : 
connect l1
connect d1

Message1 l1 -> d1 : ok no pb
d1 disconnected
Message2 l1 -> d1 (not available)
connect d1

Expected Message1 and Message2
Found Message1

**Comment by Rob174** [comment]
```
java.lang.NullPointerException: Cannot invoke "project.insa.idchatsystem.User.distanciel.User.get_id()" because "user2" is null
	at project.insa.idchatsystem.database.MessageDatabase.retrieveOrderedMessagesByConversationBetween(MessageDatabase.java:151)
	at project.insa.idchatsystem.Conversations.Conversation.Conversation.loadConversation(Conversation.java:86)
	at project.insa.idchatsystem.Conversations.Conversation.DistantConversation.<init>(DistantConversation.java:16)
	at project.insa.idchatsystem.Conversations.ConversationHandler.DistantConversationHandler.notifyNewMessage(DistantConversationHandler.java:64)
	at project.insa.idchatsystem.servlet.ServerController.notifyNewMessage(ServerController.java:51)
	at project.insa.idchatsystem.servlet.ServerSendMessage.sendPost(ServerSendMessage.java:66)
	at project.insa.idchatsystem.servlet.ServerController.sendMessage(ServerController.java:35)
	at project.insa.idchatsystem.servlet.ServerController.askForUpdate(ServerController.java:67)
	at project.insa.idchatsystem.servlet.ServerIncomingMessages.askForUpdate(ServerIncomingMessages.java:29)
	at project.insa.idchatsystem.servlet.ServerIncomingMessages.run(ServerIncomingMessages.java:18)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

**Comment by Rob174** [comment]
Tout n'est pas instancié lorsque l'on veut recevoir le msg

**Comment by Rob174** [comment]
distantconversationhandler has not the time to get all users before incoming message

**Comment by Rob174** [comment]
Solved at d750583b9de544c54737e1e055162212704c29b5
